src/tools/rustdoc-js/tester.js

```diff
@@ -190,6 +190,31 @@ function loadThings(thingsToLoad, kindOfLoad, funcToCall, fileContent) {
     return content;
 }
 
+function contentToDiffLine(key, value) {
+    return `"${key}": "${value}",`;
+}
+
+function betterLookingDiff(entry, data) {
+    let output = '{\n';
+    let error = false;
+    for (var i = 0; i < data.length && error === false; ++i) {
+        for (let key in entry) {
+            if (!entry.hasOwnProperty(key)) {
+                continue;
+            }
+            let value = data[i][key];
+            if (value !== entry[key]) {
+                output += '    - ' + contentToDiffLine(key, entry[key]) + '\n';
+                output += '    + ' + contentToDiffLine(key, value) + '\n';
+                error = true;
+            } else {
+                output += '    ' + contentToDiffLine(key, value) + '\n';
```

Comment:
This looks much better, thanks! Having a diff is really helpful :)

One thing I noticed though is that the context lines don't get the same indentation as the changed lines. Generally diffs have the same ultimate indentation for every line so that it's easier to read. Anyway, this should fix it:

```suggestion
                output += '      ' + contentToDiffLine(key, value) + '\n';
```

It may make sense to instead remove two spaces from the changed lines so that there isn't _too_ much indentation. But not really a big deal either way :)

Reply:
I'll remove the two whitespaces instead. ;)

Reply:
So this is the result:

```
Diff of first error:
{
    "path": "basic",
  - "name": "Fo",
  + "name": "Foo",
}
```

Sorry but I'll revert that change. It actually makes spotting the diff harder. :-/

Reply:
I think it would be better if you made it look like `diff` output:

```diff
 {
     "path": "basic",
-    "name": "Fo",
+    "name": "Foo",
 }

Reply:
Let me check how it looks like in the terminal (without colors, it might not be that great...).

Reply:
With the new output:

```
Diff of first error:
 {
     "path": "basic",
-     "name": "Fo",
+     "name": "Foo",
 }
```

Reply:
If it's good for both of you, I push this?

Reply:
Why do the changed lines have one space extra if indenyation though? Usually changed lines have the same indentation of context lines.

I actually preferred the style you used before, with the - and + closer to the field ;). But we can always change this later anyway.